python/src/triton.cc

```diff
@@ -1053,6 +1053,115 @@ void init_triton_ir(py::module &&m) {
               mlir::Value &rhs) -> mlir::Value {
              return mlir::Value(self.create<mlir::arith::ShRSIOp>(lhs, rhs));
            })
+      .def("create_shl_rocm",
```

Comment:
This seems very invasive. Can't this be part of the third party backend to pre-process some of the ttir?

Reply:
I will see what I can do.

Reply:
fixed